pkg/common/common.go

```diff
@@ -136,3 +138,47 @@ func RetryFunc(retryInterval, timeout time.Duration, run func() error, logger lo
 		return nil
 	}
 }
+
+func ProcessDeletePolicyForCloneGenerateRule(rules []kyverno.Rule, client *dclient.Client, p *kyverno.ClusterPolicy, logger logr.Logger) bool {
```

Comment:
Aren't `rules` and `p(policy)` duplicate here? It only uses policy name, so let's pass `name` as the string here.